schemas/v3.1/meta/base.schema.yaml

```diff
@@ -44,19 +48,17 @@ $defs:
         type: boolean
     type: object
     unevaluatedProperties: false
+dependentSchemas:
+  discriminator:
+    anyOf:
+    - required: [ oneOf ]
+    - required: [ anyOf ]
+    - required: [ allOf ]
 $dynamicAnchor: meta
 $id: https://spec.openapis.org/oas/3.1/meta/base
 $schema: https://json-schema.org/draft/2020-12/schema
 $vocabulary:
   https://spec.openapis.org/oas/3.1/vocab/base: true
-properties:
-  discriminator:
-    $ref: '#/$defs/discriminator'
-  example: true
-  externalDocs:
-    $ref: '#/$defs/external-docs'
-  xml:
-    $ref: '#/$defs/xml'
```

Comment:
I inlined them here because they aren't referenced from anywhere else, and to mirror the style we use for the standard json-schema-spec schema files. This file is more like those schemas, rather than the main openapi schema (where splitting out into definitions, even if something is only referenced once, is helpful because the document is so nested). Vocabularies aren't like that -- there's just one level: the keywords themselves.

Reply:
There is no precedent for compound keywords like these in the JSON Schema meta-schemas because there are no such keywords in JSON Schema. The only time we use objects is for map-like structures such as `properties` mapping a property name to a schema. If there were something similar in JSON Schema, I'd like to think we would use a definition for that as well.